inc/xmath.h

```diff
@@ -435,9 +434,20 @@ static inline void vec_hadamard(size_t n, TYPE dst[n], TYPE left[n], TYPE right[
 static inline TYPE vec_dot(size_t n, TYPE left[n], TYPE right[n])
 { VEC_DOT(TYPE, n, left, right) }
 
-static inline TYPE vec_mag(size_t n, TYPE left[n])
-{ VEC_MAG(TYPE, n, left) }
+static inline void mat_transpose(size_t r, size_t c, TYPE in[r][c], TYPE out[c][r])
+{ MAT_TRANSPOSE(TYPE, r, c, in, out) }
+
+static inline void mat_square_multiplication(size_t n, TYPE m[n][m_C], TYPE n[n][n], TYPE resultant[n][n]) 
+{ MAT_MUL(TYPE, n, m_C, n, n, resultant, m, n) }
 
+static inline void mat_multiplication(size_t m_R,
```

Comment:
I wonder if moving some of the args around might make this function easier to use. Something like 
m_r, m_c, M, n_r, n_c, N.

Reply:
:+1: